commodore/dependency_mgmt.py

```diff
@@ -243,6 +254,58 @@ def fetch_jsonnet_libraries(cwd: P = None):
             "the jsonnet-bundler executable `jb` could not be found"
         ) from e
 
+    # Link essential libraries for backwards compatibility.
+    lib_dir = (cwd / "vendor" / "lib").resolve()
+    lib_dir.mkdir(exist_ok=True)
+    relsymlink(
+        cwd / "vendor" / "kube-libsonnet" / "kube.libsonnet", lib_dir, "kube.libjsonnet"
+    )
+    relsymlink(
+        cwd / "vendor" / "component-argocd" / "lib" / "argocd.libjsonnet", lib_dir
+    )
+
+
+def inject_essential_libraries(file: P):
+    """
+    Ensures essential libraries are contained within `jsonnetfile.json`.
+    :param file: The path to `jsonnetfile.json`.
+    """
+    with open(file, "r") as f:
+        data = json.load(f)
+
+    deps = data["dependencies"]
+    has_kube = False
+    has_argo = False
+    for dep in deps:
+        remote = dep.get("source", {}).get("git", {}).get("remote", "")
+        has_kube = has_kube or "kube-libsonnet" in remote
+        has_argo = has_argo or "component-argocd" in remote
```

Comment:
Not sure it makes sense to add the Argo CD library like this. Shouldn't this happen via including the component only?

Reply:
Define, including the component only?

Reply:
As in "my cluster includes the Argo CD component".
I realize that this won't work for standalone component compilation. But it kind of feels wrong to me to just download the `master` of the lib if it's now already included... The current approach with faking it at least "breaks" it on purpose.

Reply:
Note, the component Argo CD gets added ONLY IF it is not already within the list of dependencies.

So if your component does define a specific version, it will not be changed to `master`.

Reply:
I dropped the Argo Component but kept kube-libsonnet. For the component compile, the Argo library still will be faked.